src/main/kotlin/com/openlattice/organizations/HazelcastOrganizationService.kt

```diff
@@ -285,44 +290,47 @@ class HazelcastOrganizationService(
         }
     }
 
+    /**
+     * Do not invoke this directly as it bypasses checks in [HazelcastOrganizationService.addMembers]
+     *
+     */
     private fun addMembers(
             orgAclKey: AclKey,
-            membersToAdd: Set<Principal>,
+            membersToAdd: Map<Principal, AclKey>,
             profiles: Map<Principal, Map<String, Set<String>>>
     ): Set<Principal> {
         require(orgAclKey.size == 1) { "Organization acl key should only be of length 1" }
-
-        val members = if (membersToAdd is Serializable) membersToAdd else membersToAdd.toSet()
-
-        val nonUserPrincipals = members.filter { it.type != PrincipalType.USER }
-        require(nonUserPrincipals.isEmpty()) { "Cannot add non-users principals $nonUserPrincipals to organization $orgAclKey" }
+        val members = membersToAdd.keys.toSet()
         val organizationId = orgAclKey[0]
-
-
         val newMembers: Set<Principal> = organizations.executeOnKey(organizationId, OrganizationEntryProcessor {
-            val newMembers = members.filter { member -> !it.members.contains(member) }.toSet()
+            val newMembers = members - it.members
             it.members.addAll(newMembers)
-            return@OrganizationEntryProcessor Result(newMembers, newMembers.isNotEmpty())
+            //We're always persisting here.
+            return@OrganizationEntryProcessor Result(newMembers)
         }) as Set<Principal>
 
-        if (newMembers.isNotEmpty()) {
-            grantOrganizationPrincipals(organizationId, newMembers, orgAclKey, profiles)
-        }
+        //Always trigger as this won't cause a write to organizations table.
+        grantOrganizationPrincipals(
+                organizationId,
+                membersToAdd,
+                orgAclKey,
+                profiles
+        )
 
         return newMembers
     }
 
     private fun grantOrganizationPrincipals(
             organizationId: UUID,
-            members: Set<Principal>,
+            members: Map<Principal, AclKey>,
             orgAclKey: AclKey = AclKey(organizationId),
             profiles: Map<Principal, Map<String, Set<String>>> = mapOf()
     ) {
         /*
          * Grant each member any of the roles, for which they meet the crieria.
          */
         val organization = organizations.getValue(organizationId)
-        members.forEach { member ->
+        members.forEach { (member, _) ->
```

Comment:
I see `_` being used below, but it wasn't defined prior to this change... how did the code compile?!

Reply:
_ is special syntactic sugar that means this argument won't be used so there is no variable _